src/Annotations/AppKernel.php

```diff
@@ -269,6 +350,70 @@ public function __construct(string $environment = 'dev')
+            $environment->addBlockRenderer(CommonMarkBlock\Element\BlockQuote::class, new CommonMark\Block\Renderer\BlockQuoteRenderer());
+            $environment->addBlockRenderer(CommonMarkBlock\Element\FencedCode::class, new CommonMark\Block\Renderer\CodeRenderer());
+            $environment->addBlockRenderer(CommonMarkBlock\Element\HtmlBlock::class, new CommonMark\Block\Renderer\HtmlBlockRenderer());
+            $environment->addBlockRenderer(CommonMarkBlock\Element\IndentedCode::class, new CommonMark\Block\Renderer\CodeRenderer());
+            $environment->addBlockRenderer(CommonMarkBlock\Element\ListItem::class, new CommonMark\Block\Renderer\ListItemRenderer());
+            $environment->addBlockRenderer(CommonMarkBlock\Element\Paragraph::class, new CommonMark\Block\Renderer\ParagraphRenderer());
+
+            $environment->addInlineRenderer(CommonMarkInline\Element\HtmlInline::class, new CommonMark\Inline\Renderer\HtmlInlineRenderer());
+            $environment->addInlineRenderer(CommonMarkInline\Element\Image::class, new CommonMark\Inline\Renderer\ImageRenderer());
+
+            return $environment;
+        };
+
+        $this->app['annotation.serializer.common_mark.doc_parser'] = function () {
+            return new DocParser($this->app['annotation.serializer.common_mark.environment']);
+        };
+
+        $this->app['annotation.serializer.common_mark.html_renderer'] = function () {
+            return new HtmlRenderer($this->app['annotation.serializer.common_mark.environment']);
+        };
+
+        $this->app['annotation.serializer'] = function () {
+            return new NormalizerAwareSerializer([
+                new AnnotationNormalizer($this->app['annotation.serializer.common_mark.doc_parser'], $this->app['annotation.serializer.common_mark.html_renderer'], $this->app['logger']),
+                new Block\CodeNormalizer(),
+                new Block\ListingNormalizer(),
+                new Block\MathMLNormalizer(),
+                new Block\ParagraphNormalizer(),
+                new Block\QuoteNormalizer(),
+                new Block\YouTubeNormalizer(),
+            ]);
+        };
+
+        $this->app['controllers.annotations'] = function () {
+            return new AnnotationsController($this->app['hypothesis.sdk'], $this->app['api.sdk'], $this->app['annotation.serializer']);
+        };
+
+        $this->app->get('/annotations', 'controllers.annotations:annotationsAction')
+            ->before($this->app['negotiate.accept'](
+                'application/vnd.elife.annotation-list+json; version=1'
+            ));
+
+        $this->app->after(function (Request $request, Response $response, Application $app) {
+            if ($response->isCacheable()) {
+                $response->headers->set('ETag', md5($response->getContent()));
+                $response->isNotModified($request);
+            }
+
+            if (!$this->app['mock'] && $this->app['debug']) {
```

Comment:
If I get this right, when `mock` is enabled during test we don't validate the application responses. Perhaps a TODO comment can be inserted to try to remove `mock` and do the validation in PHPUnit runs?

Reply:
I could get rid of this, this is not required. I found it useful in development.

Reply:
I found it useful too (to the point that I'd like it to be run also in other testing environments)